tests/test_inference_client.py

```diff
@@ -946,3 +950,32 @@ def test_token_and_api_key_mutually_exclusive(self):
     def test_model_and_base_url_mutually_exclusive(self):
         with self.assertRaises(ValueError):
             InferenceClient(model="meta-llama/Meta-Llama-3-8B-Instruct", base_url="http://127.0.0.1:8000")
+
+
+def test_stream_text_generation_response_from_bytes():
+    data = [
+        b'data: {"index":1,"token":{"id":4560,"text":" trying","logprob":-2.078125,"special":false},"generated_text":null,"details":null}',
+        b"",  # Empty line is skipped
+        b"\n",  # Newline is skipped
+        b'data: {"index":2,"token":{"id":311,"text":" to","logprob":-0.026245117,"special":false},"generated_text":" trying to","details":null}',
+        b"data: [DONE]",  # Stop signal
+        b'data: {"wont be parsed": 4}',  # Won't continue after
```

Comment:
I would reuse a previous line here, to make sure that it's not something else that's causing the parsing skipping.

Reply:
good idea :+1: